src/rgw/rgw_rest_log.cc

```diff
@@ -376,7 +376,8 @@ void RGWOp_BILog_List::execute(optional_yield y) {
 
   int shard_id;
   string bn;
-  op_ret = rgw_bucket_parse_bucket_instance(bucket_instance, &bn, &bucket_instance, &shard_id);
+  uint64_t gen_id;
+  op_ret = rgw_bucket_parse_bucket_instance(bucket_instance, &bn, &bucket_instance, &gen_id, &shard_id);
```

Comment:
let's add a separate query parameter for the generation instead of parsing it from the bucket instance